homeworks/08-functional-reactive-programming/src/hooks/useObservable.ts

```diff
@@ -0,0 +1,16 @@
+import { useState, useEffect, useMemo } from 'react';
+import { Observable, Subject, Subscription } from 'rxjs';
+
+export const useObservable = <T>(observable: Observable<T>, initial: T): T => {
+  const [value, setValue] = useState<T>(initial);
+  const subject = useMemo(() => new Subject(), []);
+
+  useEffect(() => {
+    const subscription = new Subscription();
+    subscription.add(subject);
```

Comment:
Why do you need this subscription?

Reply:
Hmm... I hope this approach should work:
```typescript
export const useObservable = <T>(observable: Observable<T>, initial: T): T => {
  const [value, setValue] = useState<T>(initial);

  useEffect(() => {
    const subscription = observable.subscribe(setValue);
    return () => subscription.unsubscribe();
  }, [observable]);

  return value;
};
```

Reply:
fixed in https://github.com/vbarbasins/evo-ts-bootcamp/pull/9/commits/06842d9ece9f65ec025c93979adeabcec0fb0c5d